task/fluxlocal/query.go

```diff
@@ -53,17 +54,31 @@ func (f *fluxQueryer) injectDependencies(ctx context.Context) context.Context {
 	}
 
 	// inject the dependencies to the context.
-	return deps.Inject(ctx)
+	return dependency.Inject(ctx, deps)
 }
 
 func (f *fluxQueryer) Query(ctx context.Context, compiler flux.Compiler) (flux.ResultIterator, error) {
 	f.logger.Info("executed flux query")
-	ctx = f.injectDependencies(context.Background())
+	var span *dependency.Span
+	ctx, span = f.injectDependencies(context.Background())
+	defer span.Finish()
 	program, err := compiler.Compile(ctx, runtime.Default)
-	alloc := &memory.Allocator{}
-	query, err := program.Start(ctx, alloc)
+	query, err := program.Start(ctx, memory.DefaultAllocator)
 	if err != nil {
 		return nil, errors.Wrap(err, "error while executing flux program")
 	}
-	return flux.NewResultIteratorFromQuery(query), nil
+	return &spannedResultIterator{
+		ResultIterator: flux.NewResultIteratorFromQuery(query),
+		span:           span,
+	}, nil
+}
+
+type spannedResultIterator struct {
+	flux.ResultIterator
+	span *dependency.Span
+}
+
+func (s *spannedResultIterator) Release() {
```

Comment:
Two questions: can this be safely called multiple times (per the Flux interface documentation for `ResultIterator.Release`)?

Should the return code from `span.Finish` be stored to be returned from the `ResultIterator.Err` interface method?  This would require implementing the method and passing back the `s.ResultIterator.Err` if non-nil, or the `s.span.Finish` return if the underlying iterator had no error.

Reply:
`Span.Finish` throws away all the errors from its nested `io.Closer`s!!!

I am shocked.